src/coreComponents/mesh/generators/CellBlockUtilities.hpp

```diff
@@ -39,17 +41,50 @@ localIndex buildEdgeMaps( localIndex numNodes,
 /**
  * @brief Get the local indices of the nodes in a face of the element.
  * @param[in] elementType Type of the element
- * @param[in] iElement the local index of the target element
- * @param[in] iFace the local index of the target face in the element  (this will be 0-numFacesInElement)
+ * @param[in] elemIdx the local index of the target element
+ * @param[in] faceNumber the local index of the target face in the element  (this will be 0-numFacesInElement)
  * @param[in] elementToNodes Element to nodes mapping.
- * @param[out] nodeIndices A reference to the array of node indices of the face. Gets resized at the proper size.
+ * @param[out] faceNodes space to write node indices to, must be of sufficient size
+ * @return number of nodes in the face
  */
-void getFaceNodes( ElementType const & elementType,
-                   localIndex const iElement,
-                   localIndex const iFace,
-                   array2d< localIndex, cells::NODE_MAP_PERMUTATION > const & elementToNodes,
-                   array1d< localIndex > & nodeIndices );
+localIndex getFaceNodes( ElementType const elementType,
+                         localIndex const elemIdx,
+                         localIndex const faceNumber,
+                         arrayView2d< localIndex const, cells::NODE_MAP_USD > const & elementToNodes,
+                         Span< localIndex > const faceNodes );
+
+/**
+ * @brief Find and count ranges of repeated values in an array of sorted arrays and compute offsets.
+ * @param [in] sortedLists the input array of sorted arrays of values
+ * @return an array of size @p sortedLists.size() + 1, where element i contains starting index of
+ *         unique values from sub-array i in a global list of unique values, while the last value
+ *         contains the total number of unique edges (i.e. an exclusive scan + total reduction).
+ */
+template< typename T >
```

Comment:
Do you think it would make sense to put add the execution policy as a template parameter. I imagine at some point down the line we may want to run as much of this initialization on the device as possible.

Reply:
Sounds good, just added it (some changes will be needed to make `forEqualRanges` host-device, as it's currently calling `std::find_if_not`, but it's easy).